### ISSUE TITLE
cmd/compile: wasm does not load successfully on 32bit X86 machine (out of memory)

### ISSUE THREAD

**Issue by dingchaoyan1983** [golang/go#34013]
### What version of Go are you using (`go version`)?

<pre>
$ go version
</pre>
go version go1.12.7

### What operating system and processor architecture are you using (`go env`)?
Windows 7 and Windows8 32bit system
<details><summary><code>go env</code> Output</summary><br><pre>
$ go env

</pre></details>

### What did you do?
loading wasm is out of memory

**Comment by ALTree** [comment]
Thanks for reporting this.

I suspect this may be related to #27462. Change [170950](https://go-review.googlesource.com/c/go/+/170950/) was merged during the 1.13 development phase and it reduces the amount of memory initially allocated by wasm code. 

I see you're on 1.12. Can you please try out the 1.13 release candidate (see https://groups.google.com/d/msg/golang-dev/EpB8KISgdEw/TOf9WHkHDgAJ) to see if it does fix your problem?

**Comment by dingchaoyan1983** [comment]
@ALTree Thanks for feedback, I have done it by you suggestion, but I have this error messages
I use the go version is `1.13rc2`
```
> GOOS=js GOARCH=wasm go build -o shared/wasm/main.wasm -i src/github.com/go-wasm/wasm.go

unexpected directory layout:
	import path: _/Users/betalpha/projects/bar-webapp/src/github.com/go-wasm/parser
	root: /Users/betalpha/projects/bar-webapp/src
	dir: /Users/betalpha/projects/bar-webapp/src/github.com/go-wasm/parser
	expand root: /Users/betalpha/projects/bar-webapp
	expand dir: /Users/betalpha/projects/bar-webapp/src/github.com/go-wasm/parser
	separator: /
```

**Comment by agnivade** [comment]
That is an unrelated issue and most probably due to where your code is placed in the filesystem. A google search gives some hits. Please look into those. Or reach out to the forums [here](https://github.com/golang/go/wiki/Questions).

And in future, please always fill the issue template in its entirety. That helps us debug the issue much faster.

**Comment by dingchaoyan1983** [comment]
I have build the wasm by version `go1.13rc2` but when I startup the wasm, the error comes
```
panic: syscall/js: call of Value.Get on undefined
go.js?6d38:22 
go.js?6d38:22 goroutine 1 [running]:
go.js?6d38:22 syscall/js.Value.Get(0x0, 0x5f296, 0x9, 0x0)
go.js?6d38:22 	/usr/local/go/src/syscall/js/js.go:252 +0x39
go.js?6d38:22 syscall.init()
go.js?6d38:22 	/usr/local/go/src/syscall/fs_js.go:21 +0xa
```
and i use the wasm_exec.js in the `misc` fold in go1.13rc2

**Comment by dingchaoyan1983** [comment]
@ALTree @agnivade the problem is resolved in version `1.13rc2`, thanks very much

**Comment by ALTree** [comment]
@dingchaoyan1983 Nice! Thank you for checking.
